src/main/java/teammates/logic/core/Emails.java

```diff
@@ -661,21 +674,63 @@ public void addEmailToTaskQueue(MimeMessage message, long emailDelayTimer) throw
         
     }
 
-    public void sendEmail(MimeMessage message) throws MessagingException {
-        log.info(getEmailInfo(message));
-        Transport.send(message);        
+    public void sendEmail(MimeMessage message) throws MessagingException, JSONException, IOException {
+        if (Config.isUsingSendgrid()) {
+            Sendgrid email = parseMimeMessageToSendgrid(message);
+            log.info(getEmailInfo(email));
+            
+            try {               
+                email.send();
+            } catch (Exception e) {
+                log.severe("Sendgrid failed, sending with GAE mail");
+                Transport.send(message);  
+            }          
+        } else {
+            log.info(getEmailInfo(message));
+            Transport.send(message);     
+        }       
     }
-    
-    
+
     /**
      * This method sends the email as well as logs its receiver, subject and content 
      * @param message
      * @throws MessagingException
+     * @throws IOException 
+     * @throws JSONException 
      */
-    public void sendAndLogEmail(MimeMessage message) throws MessagingException {
-        log.info(getEmailInfo(message));
-        Transport.send(message);
-        
+    public void sendAndLogEmail(MimeMessage message) throws MessagingException, JSONException, IOException {
+        if (Config.isUsingSendgrid()) {
+            Sendgrid email = parseMimeMessageToSendgrid(message);
+            log.info(getEmailInfo(email));
+            
+            try { 
+                email.send();           
+            } catch (Exception e) {
+                log.severe("Sendgrid failed, sending with GAE mail");
+                Transport.send(message);            
+            }
+            
+            generateLogReport(email);
+            
+        } else {
+            log.info(getEmailInfo(message));
+            Transport.send(message);           
+            generateLogReport(message);
+        }          
+    }
```

Comment:
See if you can unify this method with the `sendEmail` method above. You can have a `sendEmailWithoutLogging` and `sendEmailWithLogging` both calling a private method `sendEmail(MimeMessage msg, boolean isWithLogging)`